tests/components/fritz/test_config_flow.py

```diff
@@ -0,0 +1,416 @@
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import MOCK_CONFIG, FritzConnectionMock
+
+from tests.common import MockConfigEntry
+
+ATTR_HOST = "host"
+ATTR_NEW_SERIAL_NUMBER = "NewSerialNumber"
+
+MOCK_HOST = "fake_host"
+MOCK_SERIAL_NUMBER = "fake_serial_number"
+
+
+MOCK_USER_DATA = MOCK_CONFIG[DOMAIN][CONF_DEVICES][0]
+MOCK_DEVICE_INFO = {
+    ATTR_HOST: MOCK_HOST,
+    ATTR_NEW_SERIAL_NUMBER: MOCK_SERIAL_NUMBER,
+}
+MOCK_IMPORT_CONFIG = {CONF_HOST: MOCK_HOST, CONF_USERNAME: "username"}
+MOCK_SSDP_DATA = {
+    ATTR_SSDP_LOCATION: "https://fake_host:12345/test",
+    ATTR_UPNP_FRIENDLY_NAME: "fake_name",
+    ATTR_UPNP_UDN: "uuid:only-a-test",
+}
+
+
+@pytest.fixture()
+def fc_class_mock(mocker):
+    """Fixture that sets up a mocked FritzConnection class."""
+    result = mocker.patch("fritzconnection.FritzConnection", autospec=True)
+    result.return_value = FritzConnectionMock()
+    yield result
+
+
+async def test_user(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow by user."""
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"), patch(
+        "homeassistant.components.fritz.async_setup_entry"
+    ) as mock_setup_entry:
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+        assert result["data"][CONF_HOST] == "fake_host"
+        assert result["data"][CONF_PASSWORD] == "fake_pass"
+        assert result["data"][CONF_USERNAME] == "fake_user"
+        assert not result["result"].unique_id
+        await hass.async_block_till_done()
+
+    assert mock_setup_entry.called
+
+
+async def test_user_already_configured(hass: HomeAssistant, fc_class_mock):
+    """Test starting a flow by user with an already configured device."""
+
+    mock_config = MockConfigEntry(domain=DOMAIN, data=MOCK_USER_DATA)
+    mock_config.add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.fritz.common.FritzConnection",
+        side_effect=fc_class_mock,
+    ), patch("homeassistant.components.fritz.common.FritzStatus"):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=MOCK_USER_DATA
+        )
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
```

Comment:
Please assert the error if that's expected. Each test should finish with either a form error an abort or an entry create result.